react-time-track/src/components/user-list.js

```diff
@@ -0,0 +1,100 @@
+/** @jsx jsx */
+import React from "react";
+import { jsx } from "@emotion/core";
+
+import { Card, Circle } from "./ui";
+import { Section } from "./helpers";
+import { userList } from "../services/user";
+
+const card = {
+  background: "white",
+  borderRadius: ".5em",
+  boxShadow: "0px 2px 5px 0px rgba(0, 0, 0, .12)",
+  boxSizing: "border-box",
+  padding: "0.2rem 1rem",
+  width: "100%"
+};
+
+const circle = {
+  marginRight: "5%"
+};
+
+const eachUser = {
+  display: "flex",
+  flexDirection: "row",
+  alignItems: "center",
+  borderBottom: "1px solid gray",
+  boxSizing: "border-box",
+  padding: "0.2rem 1rem",
+  width: "100%",
+  margin: "1.2rem auto"
+};
+
+const name = {
+  width: "80%",
+  marginLeft: "5%"
+};
+
+const roleStyle = {
+  color: "gray",
+  fontSize: "18px"
+}
+
+function UserList() {
+  const [users, setUsers] = React.useState([]);
+
+  React.useEffect(() => {
+    userList().then(list => setUsers(list));
+  }, []);
+
+  let members = users.reduce((objUsers, user) => {
+    let list = objUsers[user.name[0]];
+    if (!list) list = [];
+    list.push(user.name);
+    return {
+      ...objUsers,
+      [user.name[0]]: list
+    };
+  }, {});
+
+  const roles = users.map(user => {
+    return user.role;
+  });
+  let count = -1;
+
+  const firstLetter = Object.keys(members);
+
+  return (
+    <>
+      <main>
+        <Section>
+          {firstLetter.map(user => {
+            return (
+              <>
+                <h2 css={{margin: "0px auto"}}>{user}</h2>
+                <Card styles={card} key={user}>
```

Comment:
What do you think about putting an unique key to avoid the error message in react which indicates that each child have to have a different key between among.

Reply:
Thanks. I have just corrected this section.